es6-shim.js

```diff
@@ -267,7 +267,9 @@
 
   var $String = String;
 
-  var ES = {
+  // Assign these functions to a prototype to give v8 a hint that it
```

Comment:
I can't imagine it would hurt performance, since it's just a throwaway assignment.  I'll try to double-check, but I don't know of any equivalent low-level IR tools on the firefox side.  (jsperf can be highly misleading.)